### ISSUE TITLE
Python binding static link

### ISSUE THREAD

**Issue by kyleboyle** [Hamlib/Hamlib#1554]
Is it possible to tweak the linker / compiler flags for the python binding to statically link hamlib with the python binding module so that a user would not need to install hamlib separately from the python module? I'm not super familiar with the gcc environment. 

Then using wheels for my python project I can offer the correct binary to the user.

**Comment by N0NB** [comment]
@dforsi You might already be aware of this one.

**Comment by dforsi** [comment]
@N0NB yes, I looked at this a couple of months ago but I stopped when I encountered linking errors due to mixing static an dynamic object files.

I retried just now and it compiles but the bindings aren't built so I have no idea if it works:
```
./configure --enable-shared=no --with-python-binding=yes --enable-pytest=yes
[...]
make -C bindings check
[...]
ImportError while importing test module '/home/ham/Hamlib/bindings/python/test_startup.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.13/importlib/__init__.py:88: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
python/test_startup.py:10: in <module>
    import Hamlib
Hamlib.py:12: in <module>
    import _Hamlib
E   ModuleNotFoundError: No module named '_Hamlib'
[...]
```

it build these 3 files:
```
_Hamlib.la
hamlibpy_wrap.lo
hamlibpy_wrap.o
```

@kyleboyle I suppose that you would need to link your application with those files or maybe we need a separate option to build the bindings as a shared library even if Hamlbi is built as a static library.


P.S. Having seen how many times in the WSJT-X mailing list Mike fixed issues by updating Hamlib on the fly, I think that distributing a static version of Hamlib is a bad idea that would likely increase the amount of user support needed.
